### ISSUE TITLE
chips(autocomplete): dropdown panel may display outside of visible viewport on iOS

### ISSUE THREAD

**Issue by dmatins** [angular/material#10770]
**Actual Behavior**:
 -  `What is the issue? *`
When using a select field on a mobile devise(iphone 6) the dropdown may have unexpected placement such as displaying outside of the view-able area, leaving the select field unusable and the app becomes un-scrollable.
 -  `What is the expected behavior?`
Display the dropdown for the select field directly below or above the input text.
**CodePen** (or steps to reproduce the issue): *
 -  `CodePen Demo which shows your issue:`
 -  `Details:`
Not sure how to recreate exactly but creating chips then scrolling and clicking off of the input text, then attempting to create additional chips may sometimes create the issue.

**AngularJS Versions**: *
 -  `AngularJS Version:`
1.6.2
 -  `AngularJS Material Version:`
1.1.4

**Additional Information**:
 -  `Browser Type: *` Any
 -  `Browser Version: *` Any
 -  `OS: *` IOS
 -  `Stack Traces:`

**Comment by Splaktar** [comment]
Please provide a CodePen demo with reproduction steps. This is a **required** part of our issue template.

I'm a little confused by the description because the behavior mentions `select`, but then the reproduction only mentions `chips` which only has a dropdown when combined with `autocomplete`.

I did some testing on the docs with an iPad on iOS 12 and I couldn't reproduce any issues with `md-select`.

However, testing `md-chips` used with an `md-autocomplete` gives me a lot of problems. Basically the dropdown displays below the input, but the virtual keyboard on iOS moves to just below the input. This means that the virtual keyboard covers all of the autocomplete dropdown options. This is made worse because the dropdown disables page scroll, so I can't scroll the dropdown into view.

In some cases, I was able to close the dropdown and then scroll the page up and re-open the dropdown with enough view to interact with the first element. This brought on some more unexpected behavior. If there is a list of options, I am able to scroll them with swipe up (but can still only see half of the first option). If there is only a single option, swiping up causes the background page to scroll!

I'm going to just convert this issue to focus on chips + autocomplete issues on iOS. If you are still seeing issues with `md-select` on iOS, please open a new issue (after searching for duplicates).

**Comment by Splaktar** [comment]
This is likely related to https://github.com/angular/material/issues/10970.

**Comment by garaboncias** [comment]
When a dropdown is open, only `md-virtual-repeat-container` is scrollable,
but when a dropdown has only has a few option (eg. filtered) cause scroll is not necessary inside the dropdown, user can scroll the view area when tap on an option and swipe up or down. 
Next time when a dropdown is open the layout is screwed up.

reproduction steps: 
1. open a md-autocomplete tap on an item and swipe up or down
2. open an other md-autocomplete

its not the most clear demo, but you can experience the issue:
https://codepen.io/garaboncias/full/vMoPOo

**Comment by Splaktar** [comment]
@garaboncias thank you for taking the time to provide a demo.

Today I tested on iOS 14 with an iPhone 11 Simulator and on iPadOS 14 on a real device. I wasn't able to reproduce this issue in this autocomplete-only (no chips) demo.

I also recently tested a similar issue with dialogs opening outside of the visible viewport (https://github.com/angular/material/issues/10675) and I was also unable to reproduce it on iOS 14.

So far, no one has provided the version of iOS where this can be reproduced.

Can anyone provide a demo and detailed reproduction steps for iOS/iPadOS 14?

**Comment by Splaktar** [comment]
> testing md-chips used with an md-autocomplete gives me a lot of problems. Basically the dropdown displays below the input, but the virtual keyboard on iOS moves to just below the input. This means that the virtual keyboard covers all of the autocomplete dropdown options. This is made worse because the dropdown disables page scroll, so I can't scroll the dropdown into view.

I am no longer able to reproduce these issues with chips + autocomplete in iOS 14 and AngularJS Material 1.2.0.

**Comment by Splaktar** [comment]
Not able to reproduce on a supported version of iOS.
